Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp

```diff
@@ -438,6 +515,39 @@ static void testUserContentManagerScriptMessageInWorldReceived(UserScriptMessage
     test->unregisterHandler("msg", "WebExtensionTestScriptWorld");
 }
 
+static void testUserContentManagerScriptRequestReceived(UserScriptMessageTest* test, gconstpointer)
+{
+    g_assert_true(test->registerHandler("msg", nullptr, true));
+
+    // Trying to register the same handler a second time must fail.
+    g_assert_false(test->registerHandler("msg"));
+
+    test->loadHtml("<html></html>", nullptr);
+    test->waitUntilLoadFinished();
+
+    // Check that the "window.webkit.messageHandlers" namespace exists.
+    GUniqueOutPtr<GError> error;
+    WebKitJavascriptResult* javascriptResult = test->runJavaScriptAndWaitUntilFinished("window.webkit.messageHandlers ? 'y' : 'n';", &error.outPtr());
+    g_assert_nonnull(javascriptResult);
+    g_assert_no_error(error.get());
+    GUniquePtr<char> valueString(WebViewTest::javascriptResultToCString(javascriptResult));
+    g_assert_cmpstr(valueString.get(), ==, "y");
+
+    // Check that the "document.webkit.messageHandlers.msg" namespace exists.
+    javascriptResult = test->runJavaScriptAndWaitUntilFinished("window.webkit.messageHandlers.msg ? 'y' : 'n';", &error.outPtr());
+    g_assert_nonnull(javascriptResult);
+    g_assert_no_error(error.get());
+    valueString.reset(WebViewTest::javascriptResultToCString(javascriptResult));
+    g_assert_cmpstr(valueString.get(), ==, "y");
+
+    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "'42'")));
```

Comment:
The value being sent back as result from the promise is the string `Inception` (line 332 above), so that's the value we should check for:

```suggestion
    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "Inception")));
```